test/unit/commands/skill/add-locales/helper-test.js

```diff
@@ -0,0 +1,195 @@
+        });
+
+        it('| project model intiated successfully', () => {
+            // setup
+            sinon.stub(stringUtils, 'isNonBlankString').returns(true);
+            sinon.stub(fs, 'existsSync').returns(true);
+            // call
+            expect(() => helper.initiateModels(TEST_PROFILE)).not.to.throw();
+        });
+    });
+
+    describe('# unit test for method addLocales', () => {
+        const TEST_ERROR = 'error';
+        const TEST_MODEL_PATH = 'filepath';
+        const TEST_SKILL_META_SRC = 'skillmetasrc';
+        const TEST_PATH = 'pathdoesnotmatter';
+        const TEST_TEMPLATE_BODY = { body: 'template' };
+        const TEST_TEMPLATE_MAP = {
+            statusCode: 200,
+            body: {
+                INTERACTION_MODEL_BY_LANGUAGE: {
+                    es: 'TEST_ES_URL',
+                    en: 'TEST_EN_URL'
+                }
+            }
+        };
+        new ResourcesConfig(FIXTURE_RESOURCES_CONFIG_FILE_PATH);
+        new Manifest(FIXTURE_MANIFEST_FILE_PATH);
+
+        beforeEach(() => {
+            sinon.stub(path, 'join').returns(TEST_PATH);
+            sinon.stub(httpClient, 'request');
+            sinon.stub(SkillMetadataController.prototype, 'getInteractionModelLocales');
+            sinon.stub(ResourcesConfig.prototype, 'getSkillMetaSrc').returns(TEST_SKILL_META_SRC);
+            sinon.stub(fs, 'existsSync');
+            sinon.stub(fs, 'copySync');
+            sinon.stub(fs, 'writeFileSync');
+            sinon.stub(Manifest.prototype, 'write');
+            sinon.stub(Manifest.prototype, 'getPublishingLocale').returns({ dummy: 'dummy' });
+        });
+
+        afterEach(() => {
+            sinon.restore();
+        });
+
+        it('| fail to get the template map with http error', (done) => {
+            // setup
+            httpClient.request.onFirstCall().callsArgWith(3, TEST_ERROR);
+            // call
+            helper.addLocales(['en-US'], TEST_PROFILE, TEST_DEBUG, (err, result) => {
+                expect(err).equal('Failed to retrieve the template list. Please run again with --debug to check more details.');
+                expect(result).equal(undefined);
+                done();
+            });
+        });
+
+        it('| fail to get the template map with statusCode not existing', (done) => {
+            // setup
+            httpClient.request.onFirstCall().callsArgWith(3, undefined, {});
+            // call
+            helper.addLocales(['en-US'], TEST_PROFILE, TEST_DEBUG, (err, result) => {
+                expect(err).equal('Failed to retrieve the template list. Please run again with --debug to check more details.');
+                expect(result).equal(undefined);
+                done();
+            });
+        });
+
+        it('| fail to get the template map with statusCode not correct', (done) => {
+            // setup
+            httpClient.request.onFirstCall().callsArgWith(3, undefined, { statusCode: 401 });
```

Comment:
I feel like ".yields" is much cleaner than "callsArgWith." What does 3 means for example? Hard to figure out just looking on this line. Plus, think if you add another argument, you wont need to change as many tests if you use ".yields"

Reply:
Those two are kinda similar to each other, so it's more like a personal preference I guess as to me I think define which parameter to apply the callback is clearer, so I think we can allow both ways existing, right?